testplan/web_ui/testing/src/Nav/Column.js

```diff
@@ -7,19 +7,70 @@ import {LIGHT_GREY} from "../Common/defaults";
  * Vertical column for navigation bar.
  */
 
-const Column = (props) => {
-  const leftStyle = {width: `${props.width}em`};
-  const rightStyle = {left: `${props.width}em`};
-
-  return (
-    <>
-      <div className={css(styles.column)} style={leftStyle}>
-        {props.children}
-      </div>
-      <div className={css(styles.column)} style={rightStyle}/>
-    </>
-  );
-};
+class Column extends React.Component {
+  constructor(props) {
+    super(props);
+    this.mouseEnter = this.mouseEnter.bind(this);
+    this.mouseLeave = this.mouseLeave.bind(this);
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseMove = this.mouseMove.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+
+    this.state = {
+      mouseHover: false,
+    };
+    this.enableDragging = false;
+  }
+
+  mouseEnter(e) {
+    this.setState(() => ({mouseHover: true}));
+  }
+
+  mouseLeave(e) {
+    this.setState(() => ({mouseHover: false}));
+  }
+
+  mouseDown(e) {
+    this.enableDragging = true;
+	document.addEventListener('mousemove', this.mouseMove, false);
```

Comment:
Some dodgy indentation going on here and in other places - please make sure to stick to 2-space indents. You'll need to set up your editor or IDE to set different tab widths in .js files and .py files.

Reply:
carelessly type a "tab" and it looks like 4 space in pycharm, fix it .